File: packages/es-markdown-core/src/model/link.ts
import type { MarkupBase } from "./base.js";
import type { Markup } from "./markup.js";
import type { Destination } from "../values.js";

export interface Link extends MarkupBase<"link"> {
    /**
     * Required: `[a]()` and `[a](<>)` wrote a destination and wrote nothing in
     * it, so both answer `{ kind: "url", value: "" }`. A reference occurrence
     * answers the destination its definition stated, and every occurrence of
     * one definition shares the same value.
     */
    readonly dest: Destination;
    /** Optional: `[a](/u)` wrote no title, `[a](/u "")` wrote an empty one. */
    readonly title: string | null;
    readonly content: readonly Markup[];
}

File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Link.kt
package com.nouprax.markdown.core

public class Link internal constructor(
    /**
     * Required: `[a]()` and `[a](<>)` wrote a destination and wrote nothing in
     * it, so both answer a [Destination.Url] holding `""`. A reference
     * occurrence answers the destination its definition stated.
     */
    public val dest: Destination,
    /** Optional: `[a](/u)` wrote no title, `[a](/u "")` wrote an empty one. */
    public val title: String?,
    public val content: kotlin.collections.List<Markup>,
    override val scope: Scope,
    override val anchor: String?,
    override val attributes: Attributes,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visit(this)
}
